tests/adapters/base.py

```diff
@@ -32,23 +40,92 @@ def test_shutdown(self, test_adapter):
+        #     == 3
+        # )
+        # assert (
+        #     test_adapter._parameters.get_scalar_view(
+        #         ("rf2xco2",), ("World",), "W / m^2"
+        #     ).get()
+        #     == 4
+        # )
+
+    # TODO: move this into issue about run tests
+    # def test_initialize_model_input_non_model_parameter(
+    #     self, test_adapter, test_drivers
+    # ):
+    #     tname = ("junk",)
+    #     test_adapter._parameters.get_writable_scalar_view(tname, ("World",), "K").set(4)
+    #     test_adapter.initialize_model_input()
+    #     # TODO test that "junk" has not been used
+
     def test_initialize_run_parameters(self, test_adapter, test_run_parameters):
         """
         Test that initalizing run parameters does as intended.
         """
         self.prepare_run_input(
             test_adapter, test_run_parameters.start_time, test_run_parameters.stop_time
         )
+        # TODO: uncomment these tests and check passing worked as intended
+        # expected = test_drivers["setters"]
+        # in_parameters = test_drivers["ParameterSet"]
+        # start_time = 30
+        # stop_time = 250 * 365 * 24 * 60 * 60
+        # in_parameters.get_writable_scalar_view(
+        #     ("start_time",), ("World",), "s"
+        # ).set(start_time)
+        # in_parameters.get_writable_scalar_view(
+        #     ("stop_time",), ("World",), "s"
+        # ).set(stop_time)
+
+        # out_parameters = ParameterSet()
+        # tadapter = self.tadapter(in_parameters, out_parameters)
+
         assert not test_adapter._initialized
         test_adapter.initialize_run_parameters()
         assert test_adapter._initialized
 
+        # TODO: uncomment these tests and check passing worked as intended
+        # assert tadapter._start_time == start_time
+        # assert tadapter._stop_time == stop_time
+
+        # assert (
+        #     in_parameters.get_scalar_view(("start_time",), ("World",), "s").get()
+        #     == start_time
+        # )
+        # assert (
+        #     in_parameters.get_scalar_view(("stop_time",), ("World",), "s").get()
+        #     == stop_time
+        # )
+        # assert (
+        #     in_parameters.get_scalar_view("ecs", ("World",), "K").get() == expected["ecs"]
+        # )
+        # assert (
+        #     in_parameters.get_scalar_view(
+        #         ("rf2xco2",), ("World",), "W / m^2"
+        #     ).get()
+        #     == expected["rf2xco2"]
+        # )
+        # np.testing.assert_allclose(
+        #     tadapter._parameters.get_timeseries_view(("Emissions", "CO2"), ("World",), "GtCO2/yr", expected["emissions_time_points"], ParameterType.AVERAGE_TIMESERIES).get(),
+        #     expected["emissions"],
+        #     rtol=1e-10,
+        #     atol=1e-15,
+        # )
+
     def test_run(self, test_adapter, test_run_parameters):
+        # TODO: add some tests about where input and output go to make sure adapter
```

Comment:
Enough to check "Start Time" and "Stop Time" as these are the only parameters that all models will use?

Reply:
yes that's probably a good default unless we want to make this method abstract